### DISCUSSION TITLE
Teampass 3.1.3

### DISCUSSION THREAD

**Discussion by nilsteampassnet** [nilsteampassnet/TeamPass#4508] [General]
## What's Changed
* Fix permissions issues on main.queries and users.queries. by @corentin-soriano in https://github.com/nilsteampassnet/TeamPass/pull/4400
* Fix uninitialized variables. by @corentin-soriano in https://github.com/nilsteampassnet/TeamPass/pull/4405
* Fix role visibility display issue. by @corentin-soriano in https://github.com/nilsteampassnet/TeamPass/pull/4406
* Revert excessive filter on get_teampass_settings. by @corentin-soriano in https://github.com/nilsteampassnet/TeamPass/pull/4411
* Fixing error in SQL syntax on /api/index.php/item/get?id=xxx by @nilsteampassnet in https://github.com/nilsteampassnet/TeamPass/pull/4408
* Update spanish.php by @Pruebas77 in https://github.com/nilsteampassnet/TeamPass/pull/4414
* Change password expiration calculation rules by @FengZh61 in https://github.com/nilsteampassnet/TeamPass/pull/4413
* Update license reference by @emmanuel-ferdman in https://github.com/nilsteampassnet/TeamPass/pull/4415
* Updating the items file import process by @FengZh61 in https://github.com/nilsteampassnet/TeamPass/pull/4416
* Fix the problem of abnormal values of item key and created at in item attributes by @FengZh61 in https://github.com/nilsteampassnet/TeamPass/pull/4421
* fix missing characters when exporting pdf by @FengZh61 in https://github.com/nilsteampassnet/TeamPass/pull/4429
* fix jsUrl parameter error by @FengZh61 in https://github.com/nilsteampassnet/TeamPass/pull/4428
* Fix user password expiration days error by @nilsteampassnet in https://github.com/nilsteampassnet/TeamPass/pull/4432
* Force png image type on avatar upload. by @corentin-soriano in https://github.com/nilsteampassnet/TeamPass/pull/4433
* Code review fixes by @nilsteampassnet in https://github.com/nilsteampassnet/TeamPass/pull/4412
* Fix for #4425 by @nilsteampassnet in https://github.com/nilsteampassnet/TeamPass/pull/4437
* Fix issue #4422: Resolve edit errors on previously CSV-imported items by @rcruz-pntlb in https://github.com/nilsteampassnet/TeamPass/pull/4435
* Improved robustness for csv import by @nilsteampassnet in https://github.com/nilsteampassnet/TeamPass/pull/4438
* Improve teampass cron detection. by @corentin-soriano in https://github.com/nilsteampassnet/TeamPass/pull/4446
* Exclude personal folder from identAdmin function (admins can't have personal folders). by @corentin-soriano in https://github.com/nilsteampassnet/TeamPass/pull/4447
* Fix undefined variable $results. by @corentin-soriano in https://github.com/nilsteampassnet/TeamPass/pull/4448
* Improve clickable area on the item list. by @corentin-soriano in https://github.com/nilsteampassnet/TeamPass/pull/4450
* Improve users_build_cache_tree scheduling. by @corentin-soriano in https://github.com/nilsteampassnet/TeamPass/pull/4449
* Bug report contains the oauth2 tenant id by @nilsteampassnet in https://github.com/nilsteampassnet/TeamPass/pull/4452
* Fix for oAuth2 admin page by @nilsteampassnet in https://github.com/nilsteampassnet/TeamPass/pull/4451
* Libraries update by @nilsteampassnet in https://github.com/nilsteampassnet/TeamPass/pull/4454
* Fix issue related to user allowed folder by definition by @nilsteampassnet in https://github.com/nilsteampassnet/TeamPass/pull/4439
* Only update changed rows in $tree->rebuild() to reduce load latency. by @corentin-soriano in https://github.com/nilsteampassnet/TeamPass/pull/4456
* Filter tags (keywords) values in items. by @corentin-soriano in https://github.com/nilsteampassnet/TeamPass/pull/4457
* Use random encryption key instead and reset session id on login. by @corentin-soriano in https://github.com/nilsteampassnet/TeamPass/pull/4460
* Remove login password from sessions. by @corentin-soriano in https://github.com/nilsteampassnet/TeamPass/pull/4455
* Add strong user password policy. by @corentin-soriano in https://github.com/nilsteampassnet/TeamPass/pull/4466
* Crypt session by @nilsteampassnet in https://github.com/nilsteampassnet/TeamPass/pull/4465
* Improve get_complixity_level access control. by @corentin-soriano in https://github.com/nilsteampassnet/TeamPass/pull/4467
* Update items.queries.php by @jhumphries in https://github.com/nilsteampassnet/TeamPass/pull/4463
* Updated language files with new strings by @nilsteampassnet in https://github.com/nilsteampassnet/TeamPass/pull/4468
* Remove unused endpoints in xxx.queries.php. by @corentin-soriano in https://github.com/nilsteampassnet/TeamPass/pull/4471
* Fix OTC reset on multiple account without refresh page. by @corentin-soriano in https://github.com/nilsteampassnet/TeamPass/pull/4473
* Clean XSS before html encode on csv and keepass imports. by @corentin-soriano in https://github.com/nilsteampassnet/TeamPass/pull/4474
* Fix login issue with local accounts. by @corentin-soriano in https://github.com/nilsteampassnet/TeamPass/pull/4477
* Code review by @nilsteampassnet in https://github.com/nilsteampassnet/TeamPass/pull/4469
* Improve sql queries (break long lines, use SQL COUNT and more). by @corentin-soriano in https://github.com/nilsteampassnet/TeamPass/pull/4470
* Anti bruteforce improvement by @corentin-soriano in https://github.com/nilsteampassnet/TeamPass/pull/4476
* Get user info injection by @corentin-soriano in https://github.com/nilsteampassnet/TeamPass/pull/4481
* Fix icon vertical alignment in tiems list by @nilsteampassnet in https://github.com/nilsteampassnet/TeamPass/pull/4487
* Enforce password log on copy or shown action. by @corentin-soriano in https://github.com/nilsteampassnet/TeamPass/pull/4488
* Fix backup restore process by @nilsteampassnet in https://github.com/nilsteampassnet/TeamPass/pull/4486
* Verify user password to download user keys. by @corentin-soriano in https://github.com/nilsteampassnet/TeamPass/pull/4489
* Fixed an issue that moved the item to another folder when editing it. by @corentin-soriano in https://github.com/nilsteampassnet/TeamPass/pull/4490
* Reset session before redirect to upgrade.php. by @corentin-soriano in https://github.com/nilsteampassnet/TeamPass/pull/4492
* Allow get_complexity_level in subfolders of user personal folder. by @corentin-soriano in https://github.com/nilsteampassnet/TeamPass/pull/4493
* Remove list-item-clicktoshow from icon-container. by @corentin-soriano in https://github.com/nilsteampassnet/TeamPass/pull/4494
* Anti bruteforce improvement by @corentin-soriano in https://github.com/nilsteampassnet/TeamPass/pull/4496
* Remove remaining account enumeration on login page. by @corentin-soriano in https://github.com/nilsteampassnet/TeamPass/pull/4480
* Fix items tasks error by @corentin-soriano in https://github.com/nilsteampassnet/TeamPass/pull/4498
* Prevent inconsistencies on item edition when using multiple tabs or windows. by @corentin-soriano in https://github.com/nilsteampassnet/TeamPass/pull/4499
* Allow admin to use MFA when MFA user role is defined. by @corentin-soriano in https://github.com/nilsteampassnet/TeamPass/pull/4501
* Fix install by @nilsteampassnet in https://github.com/nilsteampassnet/TeamPass/pull/4491
* Install improvements by @corentin-soriano in https://github.com/nilsteampassnet/TeamPass/pull/4503
* Allow administrators to disable user profile editing features. by @corentin-soriano in https://github.com/nilsteampassnet/TeamPass/pull/4502
* Improve condition that check if local storage and DOM are out of sync. by @corentin-soriano in https://github.com/nilsteampassnet/TeamPass/pull/4504
* Updated language files with new strings. by @nilsteampassnet in https://github.com/nilsteampassnet/TeamPass/pull/4505
* Add setting to disable drag and drop. by @corentin-soriano in https://github.com/nilsteampassnet/TeamPass/pull/4507

## New Contributors
* @Pruebas77 made their first contribution in https://github.com/nilsteampassnet/TeamPass/pull/4414
* @FengZh61 made their first contribution in https://github.com/nilsteampassnet/TeamPass/pull/4413
* @emmanuel-ferdman made their first contribution in https://github.com/nilsteampassnet/TeamPass/pull/4415
* @rcruz-pntlb made their first contribution in https://github.com/nilsteampassnet/TeamPass/pull/4435
* @jhumphries made their first contribution in https://github.com/nilsteampassnet/TeamPass/pull/4463

**Full Changelog**: https://github.com/nilsteampassnet/TeamPass/compare/3.1.2...3.1.3

<hr /><em>This discussion was created from the release <a href='https://github.com/nilsteampassnet/TeamPass/releases/tag/3.1.3'>Teampass 3.1.3</a>.</em>

**Comment by click0** [comment]
And is it possible to somehow change the minor part of the numbering of the release so that it is possible to add, for example, an update of translations to the patch release.

**Comment by poddmo** [comment]
Congratulations on the latest release and especially for growing the Teampass contributor team.
